rules/aws_provider_missing_default_tags.go

```diff
@@ -141,7 +140,8 @@ func (r *AwsProviderMissingDefaultTagsRule) Check(runner tflint.Runner) error {
 			}, nil)
 
 			if err != nil {
-				return nil
+				logger.Warn("Could not evaluate tags, skipping %s.%s.%s.%s: %s", provider.Labels[0], providerAlias, providerDefaultTagsBlockName, providerTagsAttributeName, err)
+				continue
```

Comment:
Considering that feedback may be lost in case of simple evaluation errors (e.g. undefined variables), would it be better to only short circuit in the `tflint.ErrUnknownValue` case?

Reply:
Yes, this is an option, that's what I did originally. Ignoring null variable values would be a separate case. What do you think about erroring there, versus trying to ignore it?

Reply:
I think unknown and null values should be ignored, but other errors should not be ignored. How about the following?

```go
if errors.Is(err, tflint.ErrUnknownValue) {
	logger.Warn("Could not evaluate tags, skipping %s.%s.%s.%s: %s", provider.Labels[0], providerAlias, providerDefaultTagsBlockName, providerTagsAttributeName, err)
	continue
}
```

Reply:
In spirit I agree, but the null case comes in as a raw diagnostic and not a wrapped/simple error:

https://github.com/hashicorp/hcl/blob/b4e1e146a0c3b53f973064ae3fce7b69f5ef13a0/hclsyntax/expression.go#L1219-L1230

Reply:
Oops, I missed that. As you say, if the key is null, it doesn't seem to reach `getKeysForValue`.
I misunderstood the following case:
https://github.com/terraform-linters/tflint-ruleset-aws/blob/20dacdce17a538958f8f001ebcac2a2c844733ba/rules/aws_resource_missing_tags.go#L385-L387

Reply:
Yeah it's unclear if 100% of that function is actually covered by possible configuration. Particularly this null key case. I'll write some unit tests against it and see if there's dead code.